server/views/pages/overview/partials/educationAndTrainingTabContents.test.ts

```diff
@@ -22,6 +23,33 @@ describe('Education and Training tab view', () => {
     compiledTemplate = nunjucks.compile(template.toString(), njkEnv)
   })
 
+  it('should render content', () => {
```

Comment:
I'm just considering the benefit these 'cheerio' tests give us in addition to the cypress ones? I know we have the equivalent in Spring Boot (i.e. unit and integration tests), but our cypress tests our driven via the browser, so don't they test the view inherently?

Also, remember that in Spring Boot we tend to avoid unit tests for Controllers because they contain little logic and are tested via the REST based integration tests.

Don't get me wrong, I value the benefit of high quality test coverage, but I don't want another set of tests to write and maintain, unless they add a real benefit!

Reply:
💯 You are absolutely right; and I agree 👍 
I think cypress, cheerio, and anything else we use; are just available tools in our toolbox. And our job is to use the right tool for the job 👍 

I think cheerio definitely has it's place, as do our cypress tests. So I'd be keen that we don't blindly ditch the cheerio tests we already have and remove the dependency from `package.json` (I don't think you were suggesting that)

I think the initial cheerio test I added was to introduce the library and introduce a test or two so that we could see how they are written and where they might add value etc. (to be fair I didnt make that clear though)

You are right that there is a grey line / possible overlap between the cheerio and cypress tests; and I would agree that the cypress and cheerio tests we currently have definitely overlap.

As with the Spring Boot integration tests, I would consider the cypress tests "slow", and we perhaps don't want to over do them. We don't want to do every permutation of screen flow / journey in cypress tests, just as we wouldn't with our SB integration tests. 

All the different permutations / code paths is where the unit tests sit 👍 

I see the cheerio library as part of our unit test tool box, not integration tests. With that in mind its perfect for testing any branching, looping or (possibly?) filters used in the njk templates.

As an example (and we dont have a test for this) - but the Overview page has several code branches (and I suspect there'll be more as we add more functionality):
```
Render main overview page template
If tab is Overview
    include overview tab template
If tab is Support Needs
    include support needs template
    if not problemRetrievingData
        render supper needs table
        loop through support needs
            render a row for each support need
        render selfDeclaredNeurodiversity table
        loop through selfDeclaredNeurodiversities
            render a row for each selfDeclaredNeurodiversity
        render assessedNeurodiversity table
        loop through assessedNeurodiversities
            render a row for each assessedNeurodiversity
    else
        render curious error message
If tab is Education And Training
    include education and training template
    if not problemRetrievingData
        render functional skills table
        loop through functional skills
            render a row for each functional skill
    else
        render curious error message


```

Reply:
A well reasoned response! 👍 😄